common/src/main/java/io/homeassistant/companion/android/common/data/prefs/PrefsRepositoryImpl.kt

```diff
@@ -197,4 +198,12 @@ class PrefsRepositoryImpl @Inject constructor(
     override suspend fun setIgnoredSuggestions(ignored: List<String>) {
         localStorage.putStringSet(PREF_IGNORED_SUGGESTIONS, ignored.toSet())
     }
+
+    override suspend fun getAutoFavorites(): List<String> {
+        return localStorage.getStringSet(PREF_AUTO_FAVORITES)?.toList() ?: emptyList()
+    }
+
+    override suspend fun setAutoFavorites(favorites: List<String>) {
+        localStorage.putStringSet(PREF_AUTO_FAVORITES, favorites.toSet())
```

Comment:
Sets [do not preserve order](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/-set/#:~:text=A%20generic-,unordered%20collection,-of%20elements%20that%20does%20not%20support%20duplicate%20elements.%20Methods), reordering doesn't work.

Reply:
this should be fixed now

Reply:
It isn't? If favorites start as ABC, and I change it to BAC, the next time I open the settings it's back to ABC.

Reply:
in my example I had Den Lights and Kitchen Lights.  I moved Kitchen Lights to be above Den Lights in the UI. In Android Auto it shows the order I used.

Reply:
Try reordering without adding favorites, UI doesn't necessarily reflect how it is stored because it's converting set <> list.

Reply:
Nope not working

https://github.com/home-assistant/android/assets/8148535/cfeaa977-ccb7-4a69-900f-931e9c04b78d

> if it works in practice I think its ok as of now most users will only use 1 or 2 devices here

It should work for any number of favorites, for 1 or 2 we wouldn't need reordering

Reply:
this should be fixed now in the latest commit :)